composables/useSubscriptionGraphql.ts

```diff
@@ -0,0 +1,60 @@
+import { createClient } from 'graphql-ws'
+import { getWSUrlByClient } from '@/utils/websocket'
+
+export default function ({
+  clientName = '',
+  query,
+  onChange,
+  onError,
+}: {
+  clientName?: string
+  query: string
+  onChange: (data) => void
+  onError?: (error) => void
+}) {
+  const { client: prefixClient } = usePrefix()
+
+  const client = clientName || prefixClient.value
+  const wsUrl = getWSUrlByClient(client)
+
+  if (!wsUrl) {
+    // this client do not subscription
+    return
+  }
+
+  const wsClient = createClient({
+    url: wsUrl,
+  })
+  wsClient.subscribe(
+    {
+      query: `
+          subscription {
+            ${query}
+          }
+          `,
```

Comment:
I do not honestly like this
It is really prone to error.

can we have it in the similar fashion as resolving queries?

Reply:
> can we have it in the similar fashion as resolving queries?

Could you pls expand more?
- Do you mean importing the specific .graphql file and converting it into string query ?  If so, I would use this package to implement it. https://www.npmjs.com/package/import-graphql-string.